mitodl/odl-video-service
# Collections Pagination

#### What are the relevant tickets?
Partially implements #521 

#### What's this PR do?
1. Adds pagination parameters to `getCollections` api calls
2. Adds `collectionsPagination` actions/reducers.
3. Adds `withPagedCollections` higher-order component.
4. Wraps `CollectionListPage` with `withPagedCollections`
5. Adds `Paginator` component to `CollectionListPage`.

#### How should this be manually tested?
1. Create a whole lotta collections. Here's a lil' django-shell-ism for ya:
```
from django.contrib.auth.models import User
user = User.objects.first()
from ui.factories import CollectionFactory
for i in range(500):
    collection = CollectionFactory(
        title=f'Collection_{i}',
        owner=user
    )
    collection.save()
```
2. Go to the collections page (`http://localhost:8089/collections/).
3. Confirm that the pager shows and works as you think a pager should.
4. Confirm that the sidenav always shows the first page of collections, no matter what page is active in the main page.

#### What GIF best describes this PR or how it makes you feel?
![Alt Text](https://i.gifer.com/73fS.gif)
